ls1intum/Artemis
# Add unlockAllRepositores and lockAllRepositories buttons.

### Checklist

### Motivation and Context
During an exam, some unexpected events may occur that require the instructors to intervene.
This PR adds two buttons that allow instructors to manually lock & unlock all repositories, e.g. in case the scheduling failed.

### Description
Added two new endpoints (for locking and unlocking) that will (un)lock the repositories of an exam.
The operations are performed immediately, without scheduling.
If the repositories are unlocked manually, the lock operation will still be scheduled normally.

**Note:** Locking all repositories will also lock repositories of students with extended working time!

### Steps for Testing
1. Create an exam with programming exercises
2. Register students, generate the exams and prepare the exercises (the visible date should not be reached)
3. Navigate to Bitbucket and verify that the student can not push to the repository
4. Unlock all repositories and verify that the student can push to the repository
5. Lock the repository again and verify that the student can no longer push to the repository

Please also test that when unlocking repositories manually, the lock operation is still performed automatically.

### Screenshots
Student Exams page:
![image](https://user-images.githubusercontent.com/5084100/86521624-87e95400-be53-11ea-9466-0aa87a70c18a.png)

Confirmation when unlocking the exams
![image](https://user-images.githubusercontent.com/5084100/86521628-8ddf3500-be53-11ea-837a-32dcbbbd4edd.png)

Confirmation when locking the exams
![image](https://user-images.githubusercontent.com/5084100/86521645-b0714e00-be53-11ea-8ec0-713e66b394fb.png)
